lib/restclient/utils.rb

```diff
@@ -72,23 +73,20 @@ def self._cgi_parseparam(s)
     #
     def self.cgi_parse_header(line)
       parts = _cgi_parseparam(';' + line)
-      key = parts.next
+      key = parts.shift
       pdict = {}
 
-      begin
-        while (p = parts.next)
-          i = p.index('=')
-          if i
-            name = p[0...i].strip.downcase
-            value = p[i+1..-1].strip
-            if value.length >= 2 && value[0] == '"' && value[-1] == '"'
-              value = value[1...-1]
-              value = value.gsub('\\\\', '\\').gsub('\\"', '"')
-            end
-            pdict[name] = value
+      parts.each do |p|
+        i = p.index('=')
+        if i
+          name = p[0...i].strip.downcase
+          value = p[i+1..-1].strip
+          if value.length >= 2 && value[0] == '"' && value[-1] == '"'
+            value = value[1...-1]
+            value = value.gsub('\\\\', '\\').gsub('\\"', '"')
           end
+          pdict[name] = value
         end
-      rescue StopIteration
       end
 
       [key, pdict]
```

Comment:
if an empty array was returned at line 76, this will return

```
[nil, {}]
```

Is that the desired/valid result?

Reply:
Such a result would occur when the response `Content-Type` header is missing.

I would say this is a valid result:
the `nil` item in the tuple is not consumed by the calling method (`get_encoding_from_headers`)
https://github.com/sgonzalez-r7/rest-client/blob/245178a149b19a1ce8d6b6137da044c4152d16f1/lib/restclient/utils.rb#L27

and the `get_encoding_from_headers` method will return `nil` if the 2nd tuple value is `{}` (i.e. missing `charset` in `Content-Type` header.  This is handled by its calling method (`fix_encoding`)
https://github.com/sgonzalez-r7/rest-client/blob/245178a149b19a1ce8d6b6137da044c4152d16f1/lib/restclient/response.rb#L55
in which case, the method `force_encoding` method is skipped.

and all specs pass: both unit and integration